src/test/java/com/uid2/admin/vertx/ClientKeyServiceTest.java

```diff
@@ -235,6 +235,45 @@ public void clientRolesCreatesSiteKeyIfNoneExists(Set<Role> roles, boolean siteK
         });
     }
 
+    @Test
+    public void listBySiteStringId(Vertx vertx, VertxTestContext testContext){
+        fakeAuth(Role.CLIENTKEY_ISSUER);
+
+        get(vertx, testContext, "api/client/listBySite/test", expectHttpStatus(testContext, 400));
+    }
+
+    @Test
+    public void listBySiteInvalidId(Vertx vertx, VertxTestContext testContext){
+        fakeAuth(Role.CLIENTKEY_ISSUER);
+
+        get(vertx, testContext, "api/client/listBySite/0", expectHttpStatus(testContext, 400));
+    }
+
+    @Test
+    public void listBySiteUnusedId(Vertx vertx, VertxTestContext testContext){
+        fakeAuth(Role.CLIENTKEY_ISSUER);
+
+        get(vertx, testContext, "api/client/listBySite/100", expectHttpStatus(testContext, 404));
+    }
+
+    @Test
+    public void listBySiteWorking(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.CLIENTKEY_ISSUER);
+
+        setSites(new Site(666, "test2_site", true), new Site(999, "test_site", true));
+        setClientKeys(new LegacyClientBuilder().withSiteId(999).withName("999").build(), new LegacyClientBuilder().withSiteId(666).withName("666").build());
```

Comment:
nit: I'm finding the code here a bit hard to read, could we possibly change to something like:
```
setSites(
    new Site(666, "test2_site", true),
    new Site(999, "test_site", true)
);
setClientKeys(
    new LegacyClientBuilder()
        .withSiteId(999)
        .withName("999")
        .build(),
    new LegacyClientBuilder()
        .withSiteId(666)
        .withName("666")
        .build()
);
```